kernel/thread.c

```diff
@@ -258,32 +258,57 @@ int z_impl_k_thread_name_copy(k_tid_t thread, char *buf, size_t size)
 #endif /* CONFIG_THREAD_NAME */
 }
 
-const char *k_thread_state_str(k_tid_t thread_id)
+static size_t copy_bytes(char *dest, size_t dest_size, const char *src, size_t src_size)
```

Comment:
is memcpy_s not available in our minimal libc? If not perhaps this could be placed there. This looks strikingly similar